src/components/SearchBar.vue

```diff
@@ -129,23 +134,20 @@ export default {
     },
 
     enterSearchPhrase() {
-      console.debug('Entering search phrase:', this.$data.searchPhrase);
-      if (this.$route.query.q === this.$data.searchPhrase) {
-        console.debug('No need to push the same query:', this.$data.searchPhrase);
-        return;
-      }
-      this.search({ q: this.$data.searchPhrase });
-      // We want to hide the keyboard after search has been done on Adroid
-      if (/android/i.test(navigator.userAgent)) {
-        this.hideKeyBoardOnAndroid();
+      if (this.$route.query.q !== this.searchPhraseProxy) {
+        enterSearchQuery({ q: this.searchPhraseProxy });
+        // We want to hide the keyboard after search has been done on Android
+        if (/android/i.test(navigator.userAgent)) {
+          this.hideKeyBoardOnAndroid();
+        }
       }
     },
 
     onClick() {
       // This is necessary for hiding the soft keyboard on iPhone
       // see v-closable="{ handler: 'onClick' }" in v-text-field
       // https://medium.com/@Taha_Shashtari/the-easy-vue-solution-to-dismiss-ios-keyboard-on-outside-click-2bb8be3c3347
-      this.$refs.input.blur();
+      this.$refs?.input?.blur();
```

Comment:
@reviewer: this is necessary because on (some) non-iphone devices, this causes a console error otherwise.

Reply:
ok done